slidez-core/src/main/java/com/binarystudio/academy/slidez/domain/user/model/User.java

```diff
@@ -29,24 +26,50 @@
 	@Column(name = "id", updatable = false, nullable = false)
 	private UUID id;
 
-	@Column(name = "email", unique = true)
+	@Column(name = "email", unique = true, nullable = false)
 	private String email;
 
-	@Column
-	private String firstName;
-
-	@Column
-	private String lastName;
-
-	@Column
+	@Column(name = "password", nullable = false)
 	private String password;
 
-	@Column(unique = true)
+	@Column(name = "role", nullable = false)
 	@Enumerated(EnumType.STRING)
 	private UserRole role;
 
-	public boolean isAdmin() {
-		return role == UserRole.ADMIN;
+	@OneToOne(mappedBy = "user")
+	private UserProfile userProfile;
+
+	@OneToOne(mappedBy = "user")
+	private GoogleCredentials googleCredentials;
+
+	@OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
+	@JoinColumn(name = "user_id", referencedColumnName = "id")
+	private Set<Presentation> presentations;
+
+	@Column(name = "created_at", nullable = false, updatable = false)
+	private LocalDateTime createdAt;
+
+	@Column(name = "updated_at", nullable = false)
+	private LocalDateTime updatedAt;
+
+	public User(String email, String password) {
+		this.email = email;
+		this.password = password;
+		LocalDateTime now = LocalDateTime.now();
+		this.createdAt = now;
+		this.updatedAt = now;
+	}
+
+	public static User createUser(String email, String password) {
+		User user = new User(email, password);
+		user.setRole(UserRole.USER);
```

Comment:
We must use other Role names. 
Participant and some like Presenter

Reply:
Roles might end up deleted - I don't recall any discussions about dividing responsibilities using roles. In my view, they are useless - user can be participant and presenter at the same time